client/src/visualiser-src/sorting-visualiser/data-structure/GraphicalSorts.ts

```diff
@@ -104,7 +109,132 @@ export default class GraphicalSortList extends GraphicalDataStructure {
     return producer;
   }
 
-  insertion(): AnimationProducer {
+  public merge(): AnimationProducer {
+    const producer = new SortsMergeAnimationProducer();
+
+    producer.renderMergeCode();
+
+    const low = 0;
+    const high = this.elementList.length - 1;
+
+    const p = 0;
+
+    const tmpList = [...this.elementList];
+
+    this.mergeSort(producer, low, high, low, tmpList);
+
+    return producer;
+  }
+
+  public mergeSort(producer: SortsMergeAnimationProducer, low: number, high: number, position: number, tmpList: GraphicalSortsElement[]) {
+    if (high <= low) {
+      return;
+    }
+
+    const mid = Math.floor((low + high) / 2);
+
+    let p = position;
+
+    this.mergeSort(producer, low, mid, low, tmpList);
+    this.mergeSort(producer, mid+1, high, mid+1, tmpList);
+
+    let pointerLeft = low;
+    let pointerRight = mid + 1;
+
+    producer.doAnimationAndHighlightTimestamp(
+      21,
+      false,
+      producer.highlightSorting,
+      low,
+      high,
+      this.elementList
+    );
+    
+    // compare the value on the pointerLeft and value on the pointerRight
```

Comment:
This comment is kind of wordy -- we don't really need it I think because we can tell from running the code and playing the animations.

Reply:
Thanks. I've deleted those unnecessary comments and now it should look clearer.